test/test_zany_mapping.py

```diff
@@ -100,3 +105,104 @@ def test_argyris_point(ref_cell, phys_cell):
     finat_element = finat.Argyris(ref_cell, variant="point")
     phys_element = type(finat_element.fiat_equivalent)(phys_cell, variant="point")
     check_zany_mapping(finat_element, phys_element)
+
+
+def check_zany_piola_mapping(finat_element, phys_element):
+    ref_element = finat_element._element
+    ref_cell = ref_element.get_reference_element()
+    phys_cell = phys_element.get_reference_element()
+    sd = ref_cell.get_spatial_dimension()
+    try:
+        indices = finat_element._indices
+    except AttributeError:
+        indices = slice(None)
+
+    shape = ref_element.value_shape()
+    ref_pts = make_unisolvent_points(ref_element, interior=True)
+    ref_vals = ref_element.tabulate(0, ref_pts)[(0,)*sd]
+
+    phys_pts = make_unisolvent_points(phys_element, interior=True)
+    phys_vals = phys_element.tabulate(0, phys_pts)[(0,)*sd]
+
+    # Piola map the reference elements
+    J, b = FIAT.reference_element.make_affine_mapping(ref_cell.vertices,
+                                                      phys_cell.vertices)
+    detJ = np.linalg.det(J)
+    K = J / detJ
+    if len(shape) == 2:
+        piola_map = lambda x: K @ x @ K.T
+    else:
+        piola_map = lambda x: K @ x
+
+    ref_vals_piola = np.zeros(ref_vals.shape)
+    for i in range(ref_vals.shape[0]):
+        for k in range(ref_vals.shape[-1]):
+            ref_vals_piola[i, ..., k] = piola_map(ref_vals[i, ..., k])
+
+    dofs = finat_element.entity_dofs()
+    num_bfs = phys_element.space_dimension()
+    num_dofs = finat_element.space_dimension()
+    num_facet_dofs = num_dofs - sum(len(dofs[sd][entity]) for entity in dofs[sd])
+
+    # Zany map the results
+    mappng = MyMapping(ref_cell, phys_cell)
+    Mgem = finat_element.basis_transformation(mappng)
+    M = evaluate([Mgem])[0].arr
+    shp = (num_dofs, *shape, -1)
+    ref_vals_zany = (M @ ref_vals_piola.reshape(num_bfs, -1)).reshape(shp)
+
+    # Solve for the basis transformation and compare results
+    Phi = ref_vals_piola.reshape(num_bfs, -1)
+    phi = phys_vals.reshape(num_bfs, -1)
+    Mh = np.linalg.solve(Phi @ Phi.T, Phi @ phi.T).T
+    M = M[:num_facet_dofs]
+    Mh = Mh[indices][:num_facet_dofs]
+    Mh[abs(Mh) < 1E-10] = 0.0
+    M[abs(M) < 1E-10] = 0.0
+    assert np.allclose(M, Mh), str(M.T - Mh.T)
+    assert np.allclose(ref_vals_zany[:num_facet_dofs], phys_vals[indices][:num_facet_dofs])
+
+
+@pytest.mark.parametrize("element", [
```

Comment:
cleaner/shorter if we import all the elements from finat and then save a few lines on the parameter list?